server/__tests__/loginAPI.test.js

```diff
@@ -0,0 +1,99 @@
+import { expect } from 'chai';
+import request from 'supertest';
+import { getMockUser } from '../src/utils/testUtils';
+// import User from '../src/database/models/user';
+
+let mockUser;
+
+const { testServer } = global;
+
+beforeAll(async (done) => {
+  mockUser = getMockUser();
+  // Approach 1
+  // const user = new User(mockUser);
+  // await user.hashPassword();
+  // await user.save();
+  // done();
+  // Approach 2
+  await request(testServer)
+    .post('/api/signup')
+    .send(mockUser);
+  done();
+});
+
+// /api/login test suite
+describe('Login API', () => {
+  it('returns user on logging in with email and password', (done) => {
+    const { email, password } = mockUser;
+    request(testServer)
+      .post('/api/login')
+      .send({
+        email,
+        password,
+      })
+      .expect(200)
+      .end((error, response) => {
+        const createdUser = response.body;
+        expect(createdUser).to.have.keys('id', 'username');
+        done(error);
+      });
+  });
+
+  it('gives error codes when email or password are missing', (done) => {
+    const { email } = mockUser;
+    request(testServer)
+      .post('/api/login')
+      .send({
+        email,
+      })
+      .expect(422)
+      .end((error, response) => {
+        expect(response.body).to.have.key('errors');
+        done(error);
+      });
+  });
+
+  it('gives error message when email is not found', (done) => {
+    const { password } = mockUser;
+    request(testServer)
+      .post('/api/login')
+      .send({
+        email: 'thismailcannotexist@nomailaddress.com',
+        password,
+      })
+      .expect(400)
+      .end((error, response) => {
+        expect(response.body).to.have.key('errors');
```

Comment:
is errors an array? if not, we should use error instead of errors.

Reply:
@vkbinfo , yes, errors is an array.